plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesSplitSource.java

```diff
@@ -0,0 +1,161 @@
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.deltalake.DeltaLakeErrorCode.DELTA_LAKE_INVALID_TABLE;
+import static io.trino.plugin.deltalake.functions.tablechanges.TableChangesFileSource.CDF_FILE;
+import static io.trino.plugin.deltalake.functions.tablechanges.TableChangesFileSource.DATA_FILE;
+import static io.trino.plugin.deltalake.transactionlog.TransactionLogUtil.getTransactionLogDir;
+import static io.trino.plugin.deltalake.transactionlog.checkpoint.TransactionLogTail.getEntriesFromJson;
+import static java.util.Objects.requireNonNull;
+
+public class TableChangesSplitSource
+        implements ConnectorSplitSource
+{
+    private final TrinoFileSystem fileSystem;
+    private final String tableLocation;
+    private final AtomicBoolean isFinished;
+    private AtomicLong currentVersion;
+
+    public TableChangesSplitSource(
+            ConnectorSession session,
+            DeltaLakeMetastore metastore,
+            TrinoFileSystemFactory fileSystemFactory,
+            TableChangesFunctionTableHandle functionHandle)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(functionHandle.getSchemaName(), functionHandle.getTableName());
+        this.tableLocation = requireNonNull(metastore, "metastore is null").getTableLocation(schemaTableName, session);
+        this.fileSystem = requireNonNull(fileSystemFactory, "fileSystemFactory is null").create(session);
+        currentVersion = new AtomicLong(functionHandle.getSinceVersion());
+        isFinished = new AtomicBoolean(false);
+    }
+
+    @Override
+    public CompletableFuture<ConnectorSplitBatch> getNextBatch(int maxSize)
+    {
+        long processedVersion = currentVersion.getAndIncrement();
+        Path transactionLogDir = getTransactionLogDir(new Path(tableLocation));
+        try {
+            Optional<List<DeltaLakeTransactionLogEntry>> entriesFromJson = getEntriesFromJson(processedVersion, transactionLogDir, fileSystem);
+            if (entriesFromJson.isPresent()) {
+                List<DeltaLakeTransactionLogEntry> entries = entriesFromJson.get();
+                if (!entries.isEmpty()) {
+                    List<CommitInfoEntry> commitInfoEntries = entries.stream()
+                            .filter(entry -> entry.getCommitInfo() != null)
+                            .map(DeltaLakeTransactionLogEntry::getCommitInfo)
+                            .collect(toImmutableList());
+                    checkState(commitInfoEntries.size() == 1, "There must be exactly 1 commitInfo present in a medatada file");
+                    CommitInfoEntry commitInfo = commitInfoEntries.get(0);
+                    List<ConnectorSplit> result;
+                    if (commitInfo.getOperation().equalsIgnoreCase("merge")) {
```

Comment:
Let's be a bit more inclusive here and allow dealing with other type of queries also.
Add also a TODO for https://github.com/trinodb/trino/issues/15763

Reply:
I don't understand first comment, it is not about not allowing other queries, it is about treating AddFileEntries differently for MERGE. 
Regarding second comment there is that test already: `io.trino.tests.product.deltalake.TestDeltaLakeDatabricksChangeDataFeedCompatibility#testTrinoCanReadCdfEntriesGeneratedByDatabricks`